official/vision/beta/modeling/backbones/mobilenet.py

```diff
@@ -0,0 +1,617 @@
+        ('mbconv', 5, 1, 40, 'hard_swish', 1. / 4, 6., None, False),
+        ('mbconv', 5, 1, 40, 'hard_swish', 1. / 4, 6., None, False),
+
+        ('mbconv', 5, 1, 48, 'hard_swish', 1. / 4, 3., None, False),
+        ('mbconv', 5, 1, 48, 'hard_swish', 1. / 4, 3., None, False),
+
+        ('mbconv', 5, 2, 96, 'hard_swish', 1. / 4, 6., None, False),
+        ('mbconv', 5, 1, 96, 'hard_swish', 1. / 4, 6., None, False),
+        ('mbconv', 5, 1, 96, 'hard_swish', 1. / 4, 6., None, False),
+
+        ('convbn', 1, 1, 576, 'hard_swish', None, None, True, False),
+        ('gpooling', None, None, None, None, None, None, None, None),
+        ('convbn', 1, 1, 1024, 'hard_swish', None, None, False, True),
+    ]
+}
+
+"""
+The EdgeTPU version is taken from
+github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet/mobilenet_v3.py
+"""
+MNV3EdgeTPU_BLOCK_SPECS = {
+    'spec_name': 'MobileNetV3EdgeTPU',
+    'block_spec_schema': ['block_fn', 'kernel_size', 'strides', 'filters',
+                          'activation', 'se_ratio', 'expand_ratio',
+                          'use_residual', 'use_depthwise'],
+    'block_specs': [
+        ('convbn', 3, 2, 32, 'relu', None, None, None, None),
+
+        ('mbconv', 3, 1, 16, 'relu', None, 1., True, False),
+
+        ('mbconv', 3, 2, 32, 'relu', None, 8., True, False),
+        ('mbconv', 3, 1, 32, 'relu', None, 4., True, False),
+        ('mbconv', 3, 1, 32, 'relu', None, 4., True, False),
+        ('mbconv', 3, 1, 32, 'relu', None, 4., True, False),
+
+        ('mbconv', 3, 2, 48, 'relu', None, 8., True, False),
+        ('mbconv', 3, 1, 48, 'relu', None, 4., True, False),
+        ('mbconv', 3, 1, 48, 'relu', None, 4., True, False),
+        ('mbconv', 3, 1, 48, 'relu', None, 4., True, False),
+
+        ('mbconv', 3, 2, 96, 'relu', None, 8., True, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+
+        ('mbconv', 3, 1, 96, 'relu', None, 8., False, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+        ('mbconv', 3, 1, 96, 'relu', None, 4., True, True),
+
+        ('mbconv', 5, 2, 160, 'relu', None, 8., True, True),
+        ('mbconv', 5, 1, 160, 'relu', None, 4., True, True),
+        ('mbconv', 5, 1, 160, 'relu', None, 4., True, True),
+        ('mbconv', 5, 1, 160, 'relu', None, 4., True, True),
+
+        ('mbconv', 3, 1, 192, 'relu', None, 8., True, True),
+
+        ('convbn', 1, 1, 1280, 'relu', None, None, None, None),
+    ]
+}
+
+SUPPORTED_SPECS_MAP = {
+    'MobileNetV1': MNV1_BLOCK_SPECS,
+    'MobileNetV2': MNV2_BLOCK_SPECS,
+    'MobileNetV3Large': MNV3Large_BLOCK_SPECS,
+    'MobileNetV3Small': MNV3Small_BLOCK_SPECS,
+    'MobileNetV3EdgeTPU': MNV3EdgeTPU_BLOCK_SPECS,
+}
+
+BLOCK_FN_MAP = {
+    'convbn': nn_blocks.Conv2DBNBlock,
```

Comment:
Please remove `convbn` and `gpooling` from `blocks`. They are just one or two normal layers that won't repeat or be shared in different architectures. You can just code them out plainly.

Reply:
Hey, xianzhi. As explained in the last thread. 
- I think i would keep the `BLOCK_SPECS` as it is.
- I would probably move `Conv2DBNBlock` back to `mobilenet.py` file instead of nn_blocks (but this was suggestion from @@arashwan)

HDYT?

Reply:
Sounds good to me. Thanks.